### DISCUSSION TITLE
Error : CouchDB error: deleted

### DISCUSSION THREAD

**Discussion by LTCTHITA** [Budibase/budibase#13951] [Help]
What is it? How to fix it?
 I can't save data.
![image](https://github.com/Budibase/budibase/assets/93240913/8d42e0eb-5831-4ae6-8291-154a2a1967e1)

**Comment by mikesealey** [comment]
Hey @LTCTHITA 

I'd like to look into this for you - can you provide some more details? 

**Firstly some background about your app -** 

- Are you running on cloud or self-hosted? (If self-host, specify which method)
- What version of Budibase is your app running on? (You can find your version in the Settings tab - 

![image](https://github.com/Budibase/budibase/assets/39269380/ddc0b8b7-bf99-48ea-a124-77cf777d8582)

**Next, Tell us a little about your app and what you were doing at the time this error appeared...**

- It looks like you're using the internal Budibase DB, is this correct?
- Do you also have external data sources?
- Have you recently made structural changes do your data? 
- What happens to trigger this message? (Is it displaying in the builder or the published app? Is it on page load/refresh? Are you able to publish?)
How does the data perform; does it still display in the data tab? Can it be created/updated?

**Please include as much detail as possible, and any screenshots that you feel would be useful.**

**Comment by LTCTHITA** [comment]
Hey @mikesealey 
I am changing database from BB DB to MS SQL Server.
I created a new table and changed it in the app.
I don't know if it's the right way. Do you have a better way please suggest?
![image](https://github.com/Budibase/budibase/assets/93240913/abd20494-7386-4539-8839-9f32eeea81b6)




![image](https://github.com/Budibase/budibase/assets/93240913/86173e63-490f-439c-ae77-f42e8fb5c639)
Every time you publish this notification will come up.


![image](https://github.com/Budibase/budibase/assets/93240913/d8700be2-3f81-419b-8af7-a34ffd7b7b15)
My version


I have a program upgrade, but I don't know if I can contact support because another team bought it.





![image](https://github.com/Budibase/budibase/assets/93240913/144dff1c-7432-4a55-a0c2-50b2c6406667)

**Comment by ConorWebb96** [comment]
Hey @LTCTHITA,

Did you update your data providers to work with the MySQL table? I'm assuming you re-created the table structure in MySQL and kept the everything else the exact same? It might now be coming through as null since you delete the internal table it was using previously.

Another good thing to check would be the save actions attached to any buttons. Make sure this is hooked up to the correct table. It may also be showing up as null now and thus throwing the error you are seeing when trying to save.

I hope this helps.
